tests/integration/condition__browser__have_url_test.py

```diff
@@ -19,44 +19,44 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-
-import os
-
 import pytest
 
 from selene import have
 from selene.core.exceptions import TimeoutException
-
-start_page = 'file://' + os.path.abspath(os.path.dirname(__file__)) + '/../resources/start_page.html'
+from tests.integration.helpers.givenpage import GivenPage
 
 
 def test_have_url(session_browser):
-    session_browser.open(start_page)
+    GivenPage(session_browser.driver).opened_empty()
     session_browser.should(have.url(session_browser.driver.current_url))
     session_browser.should(have.no.url(session_browser.driver.current_url[:-1]))
 
 
 def test_have_url_containing(session_browser):
-    session_browser.open(start_page)
-    session_browser.should(have.url_containing('start_page.html'))
+    GivenPage(session_browser.driver).opened_empty()
+    session_browser.should(have.url_containing('empty.html'))
     session_browser.should(have.no.url_containing('start_page.xhtml'))
 
 
 def test_fails_on_timeout_during_waiting_for_exact_url(session_browser):
     browser = session_browser.with_(timeout=0.1)
 
-    browser.open(start_page)
+    GivenPage(browser.driver).opened_empty()
 
-    with pytest.raises(TimeoutException):
+    with pytest.raises(TimeoutException) as err:
         browser.should(have.url('xttp:/'))
-        # TODO: check message too
+    assert "Timed out after 0.1s, while waiting for:" in err.value.msg
```

Comment:
this is kind of a standart part of error message, that does not depend on actual condition like have.url
so I would miss it... 
this message also helps to explain the context – that it is a "timeout error", but the latter is already reflected in with pytest.raises(TimeoutException)

Reply:
What about timeout value? Does it count here? Should we leave this message for it?

Reply:
timeout is something that we don't test in this test
and timeout is also a part of "base generic title" of all selene waiting assertions, that should be test in some different kind of test, where we focus exactly on testing "waiting logic"

Reply:
Agreed. Removed assert timeout part of msg in b2221708789e3b7a1631da963b03a107e00b382b